core/pkg/filestream/transmitchunk.go

```diff
@@ -0,0 +1,77 @@
+package filestream
+
+// TransmitChunk is data to be sent to the filestream endpoint.
+//
+// A TransmitChunk is also a CollectorStateUpdate via updating
+// the buffered data.
+type TransmitChunk struct {
+	HistoryLines    []string
+	SummaryLines    []string
+	EventsLines     []string
+	ConsoleLogLines []string
+
+	UploadedFiles []string
+
+	HasPreempting bool
+	Preempting    bool
+}
+
+func (c *TransmitChunk) Apply(state *CollectorState) {
+	state.Buffer.HistoryLines =
+		append(state.Buffer.HistoryLines, c.HistoryLines...)
+	state.Buffer.SummaryLines =
+		append(state.Buffer.SummaryLines, c.SummaryLines...)
```

Comment:
@dmitryduev should this be replaced and not appended after the change you did with summary?

Reply:
@timoffex i think based on the last change it is replace: https://github.com/wandb/wandb/pull/7553/files#diff-6983e2858eaa4bccf6d5eb4dba90ce9d30ec70cd0bff1aef3fd786d3e5682563L120-L124

unless i don't fully follow the code path

Reply:
it should be a replace, but once you have a full line...

Reply:
i can do that! so it should always be a single line, not a list?

Reply:
seems so now, since we always send the full thing, so we don't really need to append. 
I think this will probably change if we ever do the incremental updates. 

@dmitryduev we probably want to make sure that it works for longer runs, so far we only tried it with a single step.

Reply:
is there a downside to this approach in case we happen to send only a line that above the threshold, i.e. in the previous implementation we send more data, but maybe one summary will be able to be added and the next one fails. in this case we could be only sending the one that going to fail... so there is a risk here i think

Reply:
I believe both the previous implementation and this PR only send the latest line. I'm going to add a size check in `summaryupdate.go` next (which is why I factored it out like this)

Reply:
update on my last point it seems that infra always takes the last value so we don't actually change the behavior, just save on bandwith